include/seqan/blast/blast_base.h

```diff
@@ -0,0 +1,669 @@
+// sIsTranslated()
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn BlastProgramTag#sIsTranslated
+ * @brief Whether the <b>subject</b> sequence is translated in the given BlastProgram
+ * @signature constexpr bool sIsTranslated(BlastProgram const p);
+ * template <BlastProgram p>
+ * constexpr bool sIsTranslated(BlastProgramTag<p> const &);
+ * template <BlastProgram p>
+ * constexpr bool sIsTranslated(BlastProgram const _p, BlastProgramTag<p> const &);
+ *
+ * @section Remarks
+ *
+ * This function can be used to check whether the defined BlastProgam implies the
+ * nucleotide to aminoacid translation of the subject sequence.
+ *
+ * If both a tag and an enum are specified, the tag is given preference, unless it is set to
+ * @link BlastProgram::UNKNOWN @endlink.
+ *
+ * @return true for @link BlastProgram::TBLASTX @endlink and @link BlastProgram::TBLASTN @endlink
+ * @return false otherwise
+ * @headerfile <seqan/blast.h>
+ */
+
+constexpr bool
+sIsTranslated(BlastProgram const p)
+{
+    return ((p==BlastProgram::TBLASTX) ||
+            (p==BlastProgram::TBLASTN))
+            ? true
+            : false;
+}
+
+template <BlastProgram p>
+constexpr bool
+sIsTranslated(BlastProgramTag<p> const &)
+{
+    return sIsTranslated(p);
+}
+
+template <BlastProgram p>
+constexpr bool
+sIsTranslated(BlastProgram const _p, BlastProgramTag<p> const &)
+{
+    return sIsTranslated(getBlastProgram(_p, BlastProgramTag<p>()));
+}
+
+// ----------------------------------------------------------------------------
+// sNumFrames()
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn BlastProgramTag#sNumFrames
+ * @brief The number of frames per <b>query</b> sequence implied by the given BlastProgram
+ * @signature constexpr uint8_t sNumFrames(BlastProgram const p);
+ * template <BlastProgram p>
+ * constexpr uint8_t sNumFrames(BlastProgramTag<p> const &);
+ * template <BlastProgram p>
+ * constexpr uint8_t sNumFrames(BlastProgram const _p, BlastProgramTag<p> const &);
+ *
+ * @section Remarks
+ *
+ * If both a tag and an enum are specified, the tag is given preference, unless it is set to
+ * @link BlastProgram::UNKNOWN @endlink.
+ *
+ * @return 6 for @link BlastProgram::TBLASTX @endlink and @link BlastProgram::TBLASTN @endlink
+ * @return 1 otherwise
+ * @headerfile <seqan/blast.h>
+ */
+
+constexpr uint8_t
+sNumFrames(BlastProgram p)
+{
+    return (sIsTranslated(p)
+            ? 6
+            : (sHasRevComp(p)
+                ? 2
+                : 1));
+}
+
+template <BlastProgram p>
+constexpr bool
+sNumFrames(BlastProgramTag<p> const &)
+{
+    return sNumFrames(p);
+}
+
+template <BlastProgram p>
+constexpr bool
+sNumFrames(BlastProgram const _p, BlastProgramTag<p> const &)
+{
+    return sNumFrames(getBlastProgram(_p, BlastProgramTag<p>()));
+}
+
+// ----------------------------------------------------------------------------
+// _programTagToString()
```

Comment:
In general, I don't see what you are trying to do here with _programTagToString() and why it is so complicated...